src/Umbraco.Web.UI.Client/src/views/datatypes/datatype.edit.controller.js

```diff
@@ -177,26 +134,102 @@ function DataTypeEditController($scope, $routeParams, appState, navigationServic
                         err: err
                     });
 
-                    $scope.page.saveButtonState = "error";
+                    vm.page.saveButtonState = "error";
 
                     //share state
-                    editorState.set($scope.content);
+                    editorState.set(vm.content);
                 });
         }
 
     };
-
+    
+    vm.save = saveDataType;
+    
     evts.push(eventsService.on("app.refreshEditor", function(name, error) {
         loadDataType();
     }));
-
+    
     //ensure to unregister from all events!
     $scope.$on('$destroy', function () {
         for (var e in evts) {
             eventsService.unsubscribe(evts[e]);
         }
     });
 
+    function init() {
+        
+        $scope.$watch("vm.content.selectedEditor", function (newVal, oldVal) {
+            
+            //when the value changes, we need to dynamically load in the new editor
+            if (newVal && (newVal != oldVal && (oldVal || $routeParams.create))) {
+                //we are editing so get the content item from the server
+                var currDataTypeId = $routeParams.create ? undefined : $routeParams.id;
+                dataTypeResource.getPreValues(newVal, currDataTypeId)
+                    .then(function (data) {
+                        vm.preValuesLoaded = true;
+                        vm.content.preValues = data;
+                        createPreValueProps(vm.content.preValues);
+                        
+                        setHeaderNameState(vm.content);
+                        
+                        //share state
+                        editorState.set(vm.content);
+                    });
+            }
+        });
+        
+        if ($routeParams.create) {
+            
+            vm.page.loading = true;
+            vm.showIdentifier = false;
+            
+            //we are creating so get an empty data type item
+            dataTypeResource.getScaffold($routeParams.id)
+                .then(function(data) {
+
+                    vm.preValuesLoaded = true;
+                    vm.content = data;
+
+                    setHeaderNameState(vm.content);
+
+                    //set a shared state
+                    editorState.set(vm.content);
+
+                    vm.page.loading = false;
+
+                });
+        }
+        else {
+            loadDataType();
+        }
+        
+        var labelKeys = [
+            "general_settings",
+            "relations_tabName"
```

Comment:
I wonder if "Usage" could be friendlier than "Relations"?

With "Settings", my first thought was it should be "Configuration" or "Configure", (mostly because the API refers to `ConfigurationEditor`, `ConfigurationField`) - but I'm less precious about this one. ☺️

Reply:
"Usage" or "Used By" much friendlier imho.

Reply:
I had Usage before, but in some theories it should be Usages and then I didn't want to have an english war, so I chose References ... which has now been updated in JS to be consistent with c# as References.